src/dlload.c

```diff
@@ -183,13 +184,21 @@ JL_DLLEXPORT void *jl_load_dynamic_library(const char *modname, unsigned flags,
                 size_t len = strlen(dl_path);
                 if (len == 0)
                     continue;
+
+                // Is this entry supposed to be relative to the bindir?
+                if (len > 17 && strcmp(dl_path, "@executable_path/") == 0) {
```

Comment:
Is this valid if `dl_path` is just `@executable_path`?

Reply:
Ah, thanks, I had mis-read Jameson's change above.  I've reverted it now.